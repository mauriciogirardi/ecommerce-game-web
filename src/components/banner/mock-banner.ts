import { BannerProps } from '.'

export const mockBanner: BannerProps = {
  btnLabel: 'Buy now',
  btnLink: '/',
  img: '/assets/imgs/banner.png',
  title: 'Defy death',
  subtitle:
    'Play the new <strong style="color:#F231A5;">CrashLands</strong> season'
}
