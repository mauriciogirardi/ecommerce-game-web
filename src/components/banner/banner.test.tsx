import { render, screen } from '@testing-library/react'
import { describe, expect, it } from 'vitest'

import { Banner } from '.'
import { mockBanner } from './mock-banner'

describe('<Banner />', () => {
  it('should be able to render the Banner', () => {
    const { container } = render(<Banner {...mockBanner} />)

    expect(
      screen.getByRole('heading', { name: /Defy death/i, level: 1 })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', {
        name: /Play the new/i,
        level: 2
      })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', {
        name: /Play the new/i,
        level: 2
      })
    ).toBeInTheDocument()
    expect(screen.getByRole('link')).toHaveAttribute('href', '/')
    expect(screen.getByRole('img')).toHaveAttribute('alt', 'Defy death')
    expect(screen.getByText(/buy now/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
