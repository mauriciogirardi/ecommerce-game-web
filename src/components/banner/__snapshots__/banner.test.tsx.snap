// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Banner /> > should be able to render the Banner 1`] = `
<main
  class="relative md:shadow-sm"
>
  <img
    alt="Defy death"
    class="h-full w-full bg-center object-cover md:h-[580px]"
    data-nimg="1"
    decoding="async"
    height="580"
    loading="lazy"
    src="/_next/image?url=%2Fassets%2Fimgs%2Fbanner.png&w=1920&q=100"
    srcset="/_next/image?url=%2Fassets%2Fimgs%2Fbanner.png&w=750&q=100 1x, /_next/image?url=%2Fassets%2Fimgs%2Fbanner.png&w=1920&q=100 2x"
    style="color: transparent;"
    width="700"
  />
  <div
    class="bg-black/70 p-5 text-white md:absolute md:bottom-0 md:left-0 md:right-0 md:p-8"
  >
    <h1
      class="text-2xl font-bold"
    >
      Defy death
    </h1>
    <h2>
      Play the new 
      <strong
        style="color:#F231A5;"
      >
        CrashLands
      </strong>
       season
    </h2>
    <a
      class="mt-4 inline-block"
      href="/"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded text-base ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 bg-linear hover:opacity-90 h-10 px-8 text-white"
      >
        Buy now
      </button>
    </a>
  </div>
</main>
`;
